app/commands/exercise/representation/create_or_update.rb

```diff
@@ -0,0 +1,25 @@
+class Exercise
+  class Representation
+    class CreateOrUpdate
+      include Mandate
+
+      initialize_with :submission, :ast, :ast_digest, :mapping, :last_submitted_at
+
+      def call
+        representation = Exercise::Representation.create_or_find_by!(exercise:, ast_digest:) do |rep|
```

Comment:
```suggestion
        representation = Exercise::Representation.find_create_or_find_by!(exercise:, ast_digest:) do |rep|
```

To avoid empty gaps in the `id` auto increment.